projects/Mallard/src/hooks/use-response.ts

```diff
@@ -59,15 +59,15 @@ export const useResponse = <T>(
     }
 }
 
-export const withResponse = <T>(response: Response<T>) => ({
+export const withResponse = <T, R>(response: Response<T>) => ({
```

Comment:
I kind of want to add a third type here so the success and fail can be different types and then maybe do some magic so they are by default